### PR TITLE
adopt gha styling conventions

### PR DESCRIPTION
**Pull request by gmanchon**
add `publish-on-release` template from https://github.com/lewagon/challengify/blob/main/.github/workflows/publish-on-release.yml with updated style

update gha templates in order to match the style conventions enforced in https://github.com/lewagon/data-solutions/pull/889

the `clean-metadata` template is now iso with https://github.com/lewagon/data-solutions/blob/master/.github/workflows/clean-metadata.yml

the `readmes-checker` template is also iso with https://github.com/lewagon/data-solutions/blob/master/.github/workflows/readmes-checker.yml

basically once these PR are merged all referenced gha will follow the styling conventions, and the gha based on templates will use the correct yml file name and have the exact same content, which will allow them to be easily updated through the [gha store](https://github.com/lewagon/data-solutions/actions/new)

### CODE REVIEW

**Comment by gmanchon** [comment]
required by https://github.com/lewagon/utils/pull/72
